Options.py

```diff
@@ -740,7 +740,51 @@ def __str__(self) -> str:
 
     @staticmethod
     def triangular(lower: int, end: int, tri: typing.Optional[int] = None) -> int:
-        return int(round(random.triangular(lower, end, tri), 0))
+        if lower == end:
+            return lower
+
+        if lower > end:
+            # Swap the two so that `lower` is always smaller. This simplifies later code.
+            lower, end = end, lower
+
+        if tri is not None and (tri < lower or tri > end):
+            # random.triangular allows this for performance reasons, but it is not well-defined/documented behaviour, so
+            # we'll reject this scenario for simplicity.
+            raise Exception(f"Triangular distribution mode {tri} is outside the allowed range {lower}-{end}")
```

Comment:
`OptionError` doesn't seem to be used much in this file. `Range.__init__`, `Range.weighted_range` and `Range.custom_range` can all raise `Exception`, so that's why I went with `Exception` here.

Reply:
I think that's just a lack of updating to use the new error